WooCommerce/Classes/ViewRelated/Products/Add Product/ManualProductTypeOptions.swift

```diff
@@ -6,37 +6,70 @@ struct ManualProductTypeOptions: View {
+            return currentCategoryProductTypes.isEmpty ? nil : (category, currentCategoryProductTypes)
+        }
     }
 
     var body: some View {
-        ForEach(productTypes) { productType in
-            HStack(alignment: .top, spacing: Constants.margin) {
-                Image(uiImage: productType.actionSheetImage.withRenderingMode(.alwaysTemplate))
-                    .font(.title3)
-                    .foregroundStyle(.secondary)
-
-                VStack(alignment: .leading, spacing: Constants.verticalSpacing) {
-                    Text(productType.actionSheetTitle)
-                        .bodyStyle()
-                    Text(productType.actionSheetDescription)
-                        .subheadlineStyle()
+        ForEach(groupedProductTypes, id: \.0) { (category, productTypes) in
+            VStack {
+                Text(category.label)
+                    .subheadlineStyle()
+                    .textCase(.uppercase)
+                    .frame(maxWidth: .infinity, alignment: .leading)
+                    .padding(.bottom, Constants.categoryVerticalSpacing)
+                    .padding(.horizontal, Constants.horizontalSpacing)
+
+                ForEach(productTypes, id: \.self) { productType in
+                    HStack(alignment: .top, spacing: Constants.margin) {
+                        Image(uiImage: productType.actionSheetImage.withRenderingMode(.alwaysTemplate))
+                            .font(.title3)
+                            .foregroundStyle(.primary)
+                            .padding(.top, Constants.productIconTopSpacing)
+
+                        VStack(alignment: .leading, spacing: Constants.verticalSpacing) {
+                            Text(productType.actionSheetTitle)
+                                .bodyStyle()
+                            Text(productType.actionSheetDescription)
+                                .subheadlineStyle()
+                        }
+                        .padding(.bottom, Constants.productBottomSpacing)
+                        Spacer()
+                    }
+                    .padding(.horizontal, Constants.horizontalSpacing)
+                    .onTapGesture {
+                        onOptionSelected(productType)
+                    }
+
+                }
+                if category != Constants.productCategoriesOrder.last {
+                    Divider()
+                        .padding(.vertical, Constants.categoryVerticalSpacing)
                 }
-                Spacer()
-            }
-            .onTapGesture {
-                onOptionSelected(productType)
             }
         }
     }
 }
 
 private extension ManualProductTypeOptions {
     enum Constants {
+        // List of product categories. The ordering dictates how the categories are displayed.
+        static let productCategoriesOrder: [BottomSheetProductType.ProductCreationCategory] = [.standard, .subscription, .other]
```

Comment:
Suggestion: You can omit this if you make `BottomSheetProductType.ProductCreationCategory` conform to `CaseIterable`. Then you can get the list of cases in the order declared in the enum with `BottomSheetProductType.ProductCreationCategory.allCases`.

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/12273/commits/f1b37ed71271104480c29d1be2724a266c109d5b  also this comment is related to this https://github.com/woocommerce/woocommerce-ios/pull/12273#discussion_r1634539342